ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadManager.swift

```diff
@@ -74,6 +75,9 @@ public class ResourceDownloadManager {
       // Perform common 'can download' check.  We need positive reachability and no prior download queue.
       // The parameter facilitates error logging.
       if !downloader.canExecute(.simpleBatch(dlBatch)) {
+        let error = NSError(domain: "Keyman", code: 0,
+                  userInfo: [NSLocalizedDescriptionKey: "Download queue is either busy or lacks internet connection"])
```

Comment:
Is this a message for the end user? I have no idea what it means -- nor what action could be taken as a result! 😁 

(BTW, some smells slightly WET here...)

Reply:
Upon inspection, `downloadKeyboardCore` and `downloadLexicalModelCore` are near-identical, aside from the following:
- `downloadKeyboardCore` takes one extra parameter, which is only forwarded to...
- calls `buildKeyboardDownloadBatch` vs `buildLexicalModelDownloadBatch`.

The called methods are currently a bit distinct in what they do, but that should end once we're downloading KMPs for keyboards.  That'll be the best time to nix the WETness seen here.

Reply:
I went ahead and retooled the check a bit.  `ResourceDownloadQueue` now implements `.state: QueueState` - an `enum` - instead of `.canExecute() -> Bool`.  The new enum also provides common error definitions, reinstating the original error messages that `.canExecute` used to return directly.